src/S3Commands.hh

```diff
@@ -27,19 +27,20 @@
 class AmazonRequest : public HTTPRequest {
   public:
 	AmazonRequest(const S3AccessInfo &ai, const std::string objectName,
-				  XrdSysError &log)
+				  XrdSysError &log, bool ro = true)
 		: AmazonRequest(ai.getS3ServiceUrl(), ai.getS3AccessKeyFile(),
 						ai.getS3SecretKeyFile(), ai.getS3BucketName(),
 						objectName, ai.getS3UrlStyle(),
-						ai.getS3SignatureVersion(), log) {}
+						ai.getS3SignatureVersion(), log, ro) {}
 
 	AmazonRequest(const std::string &s, const std::string &akf,
 				  const std::string &skf, const std::string &b,
 				  const std::string &o, const std::string &style, int sv,
-				  XrdSysError &log)
+				  XrdSysError &log, bool ro = true)
```

Comment:
This constructor change doesn't appear to be used.  Let's scrap it.

Reply:
This constructor is used by the courtesy constructor that has a simpler interface, does some translations and then calls this one. This is actually where the bookkeeping is done for the retention policy.